x-pack/plugins/cases/public/components/templates/utils.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { getConnectorsFormSerializer, isEmptyValue } from '../utils';
+import type { TemplateFormProps } from './types';
+
+export const removeEmptyFields = (
+  data: TemplateFormProps | Record<string, string | boolean | null | undefined> | null | undefined
+): TemplateFormProps | Record<string, string | boolean> | null => {
+  if (data) {
+    return Object.entries(data).reduce((acc, [key, value]) => {
+      let initialValue = {};
+
+      if (key === 'customFields') {
+        const nonEmptyCustomFields =
+          removeEmptyFields(value as Record<string, string | boolean | null | undefined>) ?? {};
+
+        if (Object.entries(nonEmptyCustomFields).length > 0) {
+          initialValue = {
+            customFields: nonEmptyCustomFields,
+          };
+        }
+      } else if (!isEmptyValue(value)) {
+        initialValue = { [key]: value };
+      }
+
+      return {
+        ...acc,
+        ...initialValue,
+      };
+    }, {});
+  }
+
+  return null;
+};
+
+export const templateSerializer = <T extends TemplateFormProps | null>(data: T): T => {
+  if (data !== null) {
+    const { fields = null, ...rest } = data;
+    const connectorFields = getConnectorsFormSerializer({ fields });
+    const serializedFields = removeEmptyFields({ ...rest, fields: connectorFields.fields });
+
+    return {
+      ...serializedFields,
+      // fields: connectorFields.fields,
```

Comment:
Had to uncomment it because improved `removeEmptyFields` function removes everything which is null or empty. 
However if few of the connectorFields (e.g.  impact: '1', severity: '2', category: null) have value then we need all the  connector fields to exist (event null fields).